### PR TITLE
Fixes #2497 - Join to the fact_values table when a user fact-filter is applied

### PR DESCRIPTION
**Pull request by GregSutcliffe**
There appear to be no tests at all for the user fact filtering, so I've added some. These tests fail correctly without the change to app/...

### CODE REVIEW

**Comment by ohadlevy** [comment]
merged at 9ae88fc  thanks!

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -131,7 +131,9 @@ class Jail < ::Safemode::Jail
       conditions.sub!(/^(?:\(\))?\s?(?:and|or)\s*/, "")
       conditions.sub!(/\(\s*(?:or|and)\s*\(/, "((")
     end
-    {:conditions => conditions}
+
+    { :joins => ufs.empty? ? nil : :fact_values,
+      :conditions => conditions }
```
Comment (dmitri-d) [inline app/models/host/managed.rb:136]: { :joins => :fact_values, :conditions => conditions } unless ufs.empty? 

would be a bit better, IMO.

File: test/unit/host_test.rb [hunk 2]
```diff
@@ -160,6 +160,17 @@ def setup_user_and_host
...
+    # Can't use `setup_user_and_host` as it deletes the UserFacts
+    @one             = users(:one)
+    @one.hostgroups  = []
+    @one.domains     = []
+    @one.user_facts  = [user_facts(:one)]
+    @one.facts_andor = "and"
+    @one.save!
+    User.current    = @one
+  end
+
```
Comment (dmitri-d) [inline test/unit/host_test.rb:173]: I tend to prefer update_attributes! to explicit assignments and subsequent saves, probably just a personal style though.

File: app/models/host/managed.rb (hunk 1 above)
Comment (GregSutcliffe) [inline app/models/host/managed.rb:136]: Indeed. I suspect this also needs a grouping clause as well, I _think_ I saw it once return 63 rows of the same host ;)

**Pull request closed without merging**